parcel/ParcelDistributions.jl

```diff
@@ -17,16 +23,22 @@ function distribution_moments(::Monodisperse, qₗ, Nₗ, ρₗ, ρ_air, qᵢ, N
         Aₗ = 4 * FT(π) * rₗ^2
         Vₗ = FT(4 / 3 * π) * rₗ^3
     end
+    return (; rₗ, Aₗ, Vₗ)
+end
+
+function distribution_moments(::Monodisperse, ::TD.Ice, qᵢ, Nᵢ, ρᵢ, ρ_air)
+    FT = typeof(qᵢ)
     # ice crystals
-    if Nᵢ == FT(0)
+    if Nᵢ == FT(0) || qᵢ == FT(0)
         rᵢ = FT(0)
     else
         rᵢ = cbrt(qᵢ / Nᵢ / FT(4 / 3 * π) / ρᵢ * ρ_air)
     end
-    return (; rₗ, Aₗ, Vₗ, rᵢ)
+    return (; rᵢ)
 end
 
-function distribution_moments(::Gamma, qₗ, Nₗ, ρₗ, ρ_air, qᵢ, Nᵢ, ρᵢ)
+function distribution_moments(::Gamma, ::TD.Liquid, qₗ, Nₗ, ρₗ, ρ_air)
```

Comment:
We are passing in liquid and ice densities as input. So I don't think we need to have a separate second argument for the condensate type. `function distribution_moments(::Gamma, q, N, ρ, ρ_air)` should work for both liquid and ice. And I think the same for other distributions. And then we would just need to call it twice in parcel models to specify separate distributions for liquid droplets and ice crystals